Mathlib/Data/Fintype/Sum.lean

```diff
@@ -24,10 +24,38 @@ instance (α : Type u) (β : Type v) [Fintype α] [Fintype β] : Fintype (α ⊕
   elems := univ.disjSum univ
   complete := by rintro (_ | _) <;> simp
 
-@[simp]
-theorem Finset.univ_disjSum_univ {α β : Type*} [Fintype α] [Fintype β] :
-    univ.disjSum univ = (univ : Finset (α ⊕ β)) :=
-  rfl
+namespace Finset
+variable {α β : Type*}
+
+section left
+variable [Fintype α] {u : Finset (α ⊕ β)}
+
+@[simp] lemma toLeft_eq_univ : u.toLeft = univ ↔ univ.disjSum ∅ ⊆ u := by
```

Comment:
My instinct is that this is the wrong way round for simp: the right hand side is a fair bit more complex than the left. Similarly for the lemmas below.

Reply:
I'd say "no" because `univ.disjSum ∅` is a fixed finset which might as well be given a name

Reply:
Ah, I see what's going on. These lemmas aren't generalised correctly, you instead want to be proving
```
@[simp] lemma disjSum_subset_iff :
    s.disjSum t ⊆ u ↔ s ⊆ u.toLeft ∧ t ⊆ u.toRight := by
  simp [Finset.subset_iff]

@[simp] lemma subset_disjSum_iff :
    u ⊆ s.disjSum t ↔ u.toLeft ⊆ s ∧ u.toRight ⊆ t := by
  simp [Finset.subset_iff]
```
then both of your lemmas are provable by simp in the opposite direction. Trust the universal property!